autogen/header.rs

```diff
@@ -177,6 +200,20 @@ pub fn gen_spirv_header(grammar: &structs::Grammar) -> TokenStream {
         pub enum Op {
             #(#opcodes),*
         }
+
+        impl num_traits::FromPrimitive for Op {
```

Comment:
is there a way to avoid repeating this block over and over in autogen?

Reply:
How? This is a different `FromPrimitive` impl for every generate enum.

Reply:
some helper function that generates `TokenStream` with this implementation based on the type and the list of enum variants that we provide?